### DISCUSSION TITLE
v0.10.0

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#504] [Announcements]
这是 Magpie 诞生以来最重大的更新，感谢每一位贡献代码、翻译以及提供反馈的人。

感谢 @yakymoto 和 @0x4E69676874466F78 分别贡献了英语和俄语翻译。

**运行环境变更**
1. 不再支持 Win10 v1903 以下的系统
2. 不再需要 .NET 和 MSVC 运行时

**新功能**
1. 全新的基于 WinUI 的用户界面，支持明暗主题和 Mica 背景
    * Mica 背景仅限 Win11 22H2+
 2. 特定窗口（游戏）的配置 #423 #385 #130 #61
    * 可直接在配置界面启动游戏
    * 支持自动缩放
3. 支持在用户界面中配置缩放模式
4. 支持“默认降采样效果”，窗口缩放后如果大于屏幕尺寸将自动应用此效果
    * 默认为 Bicubic
5. 支持导出/导入缩放模式，可用于分享配置
    * 支持导入旧版缩放配置（ScaleModels.json）
6. 新的修改快捷键的用户界面，和 PowerToys 体验相同
7. 便携模式，开启后会将配置文件保存在本地文件夹
8. 升级了常驻后台功能，开启后始终在系统托盘中显示图标，关闭主窗口不会退出程序
9. 自动更新 #287
10. 开机启动时不会弹出 UAC #327
11. 支持自定义倒计时时长 #258
12. 系统截图时自动隐藏光标 #394
    * 卡顿时可能不可靠
13. “保留标题栏”选项
    * 仅限 Graphics Capture 和 Desktop Duplication 捕获模式
14. 支持自定义光标缩放
15. “内联效果参数”选项。开启后将稍微提高性能，但每次修改参数将需重新编译该效果
16. 源窗口移动或调整大小后自动重新缩放 #487

**修复**
1. 修复了与 RTSS 的兼容性
2. 优化了对透明窗口的支持 #422
3. 现在禁止缩放操作系统的特殊窗口，如开始菜单、任务栏、桌面窗口等
4. 现在缩放时可以被 Wallpaper Engine 检测到 #502

**效果变更**
1. 添加 Deband #472 #479 （感谢 @kato-megumi）
2. CRT_Geom 默认不再模拟隔行扫描 #415
3. 部分基于机器学习的效果现在有了更平滑的边缘 #398
4. 删除 CatmullRom
5. 修复了 SMAA 错误的支持缩放的问题

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.10.0'>v0.10.0</a>.</em>

**Comment by hooke007** [comment]
用管理员启动一次之后退出后，再也无法启动了
[magpie.log](https://github.com/Blinue/Magpie/files/11069419/magpie.log)
appdata目录的配置清理后仍无法启动

**Reply by hooke007** [reply]
找到复现方式了。
用管理员启动，设置里勾选“始终以管理员启动”，退出。
启动magpie，无响应（实际上任务管理器有个magpie进程），手动结束后再以管理员启动。

**Reply by Blinue** [reply]
有没有打开“在系统托盘上显示应用程序”？打开的话必须用托盘菜单退出

**Reply by hooke007** [reply]
自然是打开的，而且此时也只能托盘菜单退出

**Reply by hooke007** [reply]
没有图标
![image](https://user-images.githubusercontent.com/41094733/227756143-998651ed-d244-4d90-af12-80e4bcd6d49e.png)

**Reply by Blinue** [reply]
似乎托盘图标创建失败了？提一个 issue 吧

**Comment by yakymoto** [comment]
Just want to say THANK YOU for this awesome release.
Keep up the good work!

**Comment by NeilTohno** [comment]
感谢巨巨，ヽ(ﾟ∀ﾟ)ﾒ(ﾟ∀ﾟ)ﾉ 

分享我自己写的配置，看文字说明
https://github.com/NeilTohno/Magpie/releases/tag/v0.10.0

[![Sge0F.th.png](https://imglink.win/image/2023/03/29/Sge0F.th.png)](https://imglink.org/image/Sge0F)

**Comment by Crachlow** [comment]
Hello Blinue. Faced such a problem. If it's possible, how can I fix it. FSR - Desktop Duplication
https://ibb.co/BV8yg1W

**Reply by hooke007** [reply]
https://github.com/Blinue/Magpie/wiki/FAQ%20(EN)#duplicate-monitoring

**Comment by hooke007** [comment]
我在折腾的时候发现了两个小点：

1. 缩放模式允许重名是不是有点奇怪 <img src="https://user-images.githubusercontent.com/41094733/231256876-d53c44af-37d6-4ea3-8454-f37ed8a9be9d.jpg" height=300>
2. 导入缩放配置的时候是不是要加个选择，“覆盖”或“追加”。误操作导入了两次大的配置，看着几十条瞬间麻了，一条条手动删是不可能的，只能直接删配置文件了

**Reply by Blinue** [reply]
第一点限制起来有些麻烦，我倾向于不做限制，取什么名字由用户决定，内部使用索引标识。
第二点的话我有一个想法，以后在导入/导出时可以选择要导入/导出的缩放模式，也可以选择覆盖还是追加。
